src/peft/helpers.py

```diff
@@ -146,3 +148,20 @@ def check_if_peft_model(model_name_or_path: str) -> bool:
         is_peft_model = False
 
     return is_peft_model
+
+
+@contextmanager
+def set_adapter_scale(model, alpha):
+    # 1. TODO: Check whether scaling is prohibited on model
+
+    # 2. Modify scaling values
+    original_scaling = {}
+    for module in model.modules():
+        if isinstance(module, LoraLayer):
+            original_scaling[module] = module.scaling.copy()
+            module.scaling = {k: v * alpha for k, v in module.scaling.items()}
+    yield
```

Comment:
Let's use
```python
try:
    yield
finally:
    # restore
```

That way it is ensured that the original scale is restored, even if there is an error in-between. There should be a test for this edge case too.

Reply:
This makes sense to me! Thanks for the suggestion.